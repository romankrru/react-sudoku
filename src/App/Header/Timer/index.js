import React, { Fragment } from "react";

import styles from "./index.module.css";

const formatTime = (seconds) => {
	const ss = seconds % 60;
	const mm = Math.floor(seconds / 60) % 60;
	const hh = Math.floor(seconds / (60 * 60));
	const toStr = (t) => t.toString().padStart(2, "0");

	return [hh, mm, ss].map(toStr);
};

const Timer = (props) => {
	const [hh, mm, ss] = formatTime(props.seconds);

	return (
		<div className={styles.timer}>
			<div className={styles.timerClock}>
				{hh !== "00" && (
					<Fragment>
						{hh}
						<span className={styles.timerLegend}>h</span>{" "}
					</Fragment>
				)}

				{mm !== "00" && (
					<Fragment>
						{mm}
						<span className={styles.timerLegend}>min</span>{" "}
					</Fragment>
				)}

				<Fragment>
					{ss}
					<span className={styles.timerLegend}>sec</span>
				</Fragment>
			</div>

			<div className={styles.timerControl}></div>
		</div>
	);
};

export default Timer;
