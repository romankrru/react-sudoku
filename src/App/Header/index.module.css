.header {
    height: 300px;
    background-color: #2979FA;
}

.headerContainer {
    position: relative;

    width: 450px;
    margin: 0 auto;
}

.headerTop {
    display: flex;
    justify-content: space-between;
    align-items: center;
    margin-bottom: 15px;
}

.headerControls {
    display: flex;
    flex-direction: row;
}

.headerControlsButton {
    width: 100%;
    margin-right: 30px;
}

.headerControlsButton:last-child {
    margin-right: 0;
}

.difficultyControls {
    position: absolute;
    top: 160px;
    left: 475px;
}

.difficultyRadio {
    margin-bottom: 6px;
}