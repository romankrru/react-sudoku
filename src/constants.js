const difficulty = {
	EASY: "EASY",
	MEDIUM: "MEDIUM",
	HARD: "HARD",
};

const modes = {
	NORMAL: "NORMAL",
	CANDIDATE: "CANDIDATE",
};

const EMPTY_CELL = 0;

export { difficulty, modes, EMPTY_CELL };
