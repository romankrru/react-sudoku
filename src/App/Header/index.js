import React from "react";

import Button from "../../generic/Button";
import Radio from "../../generic/Radio";
import { difficulty } from "../../constants";
import Timer from "./Timer";
import styles from "./index.module.css";
import logo from "./assets/logo.svg";

const Header = (props) => (
	<header className={styles.header}>
		<div className={styles.headerContainer}>
			<div className={styles.headerTop}>
				<img src={logo} alt="Sudoku" width="149" height="80" />
				<Timer
					seconds={props.seconds}
					toggle={props.toggleTimer}
					isRunning={props.isTimerRunning}
				/>
			</div>
			<div className={styles.headerControls}>
				<Button className={styles.headerControlsButton}>New</Button>
				<Button onClick={props.reset} className={styles.headerControlsButton}>
					Reset
				</Button>
				<Button className={styles.headerControlsButton}>Solve</Button>
			</div>
			<form className={styles.difficultyControls}>
				<Radio
					name="difficulty"
					value={difficulty.EASY}
					className={styles.difficultyRadio}
				>
					Easy
				</Radio>
				<Radio
					name="difficulty"
					value={difficulty.MEDIUM}
					className={styles.difficultyRadio}
				>
					Medium
				</Radio>
				<Radio
					name="difficulty"
					value={difficulty.HARD}
					className={styles.difficultyRadio}
				>
					Hard
				</Radio>
			</form>
		</div>
	</header>
);

export default Header;
