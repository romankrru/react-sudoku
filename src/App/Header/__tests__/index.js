import React from "react";
import { render, screen } from "@testing-library/react";

import Header from "../";
import { difficulties } from "../../../constants";

it("<Header /> renders and has logo with alt text", () => {
	const view = render(
		<Header
			seconds={12}
			newGame={jest.fn}
			reset={jest.fn}
			setDifficulty={jest.fn}
			toggleTimer={jest.fn}
			isTimerRunning={true}
			difficulty={difficulties.EASY}
		/>
	);

	expect(screen.getByAltText("Sudoku")).toBeInTheDocument();
	expect(view.container).toMatchSnapshot();
});
