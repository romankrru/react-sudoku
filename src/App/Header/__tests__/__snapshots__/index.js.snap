// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders and has logo with alt text 1`] = `
<div>
  <header
    class="header"
  >
    <div
      class="headerContainer"
    >
      <div
        class="headerTop"
      >
        <img
          alt="Sudoku"
          height="80"
          src="logo.svg"
          width="149"
        />
        <div
          class="timer"
        >
          <div
            class="timerClock"
          >
            12
            <span
              class="timerLegend"
            >
              sec
            </span>
          </div>
          <div
            class="timerControl timerControlPause"
          />
        </div>
      </div>
      <div
        class="headerControls"
      >
        <button
          class="button headerControlsButton"
        >
          New
        </button>
        <button
          class="button headerControlsButton"
        >
          Reset
        </button>
      </div>
      <form
        class="difficultyControls"
      >
        <label
          class="radio difficultyRadio"
        >
          <input
            checked=""
            class="input"
            name="difficulty"
            type="radio"
            value="EASY"
          />
          <div
            class="indicator"
          />
          Easy
        </label>
        <label
          class="radio difficultyRadio"
        >
          <input
            class="input"
            name="difficulty"
            type="radio"
            value="MEDIUM"
          />
          <div
            class="indicator"
          />
          Medium
        </label>
        <label
          class="radio difficultyRadio"
        >
          <input
            class="input"
            name="difficulty"
            type="radio"
            value="HARD"
          />
          <div
            class="indicator"
          />
          Hard
        </label>
      </form>
    </div>
  </header>
</div>
`;
